packages/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/launcher/loaders/DevLauncherAppLoader.kt

```diff
@@ -94,4 +96,59 @@ abstract class DevLauncherAppLoader(
   private fun launchIntent(intent: Intent) {
     context.applicationContext.startActivity(intent)
   }
+
+  private fun maybeInitFlipper(appContext: Context, reactInstanceManager: ReactInstanceManager) {
+    try {
+      val wasRunning = tryToStopFlipper()
+
+      // Flipper wasn't initialized in the MainApplication so we don't want to start it.
+      if (!wasRunning) {
+        return
+      }
+
+      /*
+      We use reflection here to pick up the class that initializes Flipper,
+      since Flipper library is not available in release mode
+      */
+      val packageName = appContext.packageName
+      val aClass = Class.forName("$packageName.ReactNativeFlipper")
+      aClass
+        .getMethod("initializeFlipper", Context::class.java, ReactInstanceManager::class.java)
+        .invoke(null, context, reactInstanceManager)
+    } catch (e: ClassNotFoundException) {
+      e.printStackTrace()
+    } catch (e: NoSuchMethodException) {
+      e.printStackTrace()
+    } catch (e: IllegalAccessException) {
+      e.printStackTrace()
+    } catch (e: InvocationTargetException) {
+      e.printStackTrace()
+    }
+  }
+
+  private fun tryToStopFlipper(): Boolean {
+    val androidFlipperClientClass = Class.forName("com.facebook.flipper.android.AndroidFlipperClient");
+    val getInstanceMethod = androidFlipperClientClass.getMethod("getInstanceIfInitialized")
```

Comment:
Do you think it's worth adding a separate try-catch block around here, or elsewhere in this method, so that if these method/field names change in the future we will still reach the `initializeFlipper` call?

Reply:
I think we should disable this workaround when one of the methods in that method fails. Staring a flipper when one instance is already running doesn't have any effect, but what if the flipper was stope but plugins weren't removed. It's hard to predict what will happen. By not adding the `try-catch` block here, we will only use a workaround when we're certain that it will work.